crates/sui-framework/packages/sui-system/sources/validator_set.move

```diff
@@ -504,20 +557,65 @@ fun update_and_process_low_stake_departures(
             // If the grace period has passed, the validator has to leave us.
             if (new_low_stake_period > low_stake_grace_period) {
                 let validator = self.active_validators.remove(i);
-                process_validator_departure(self, validator, validator_report_records, false /* the validator is kicked out involuntarily */, ctx);
+                let removed_stake = self.process_validator_departure(
+                    validator,
+                    validator_report_records,
+                    false /* the validator is kicked out involuntarily */,
+                    ctx
+                );
+                total_removed_stake = total_removed_stake + removed_stake;
             }
+        // SIP-39: at the end of an epoch when new voting powers are computed based on stake changes,
+        //      any validator with VOTING_POWER < VERY_LOW_VOTING_POWER_THRESHOLD will be removed
```

Comment:
similar to my previous comment - it seems that a validator can be kicked out due to pending stake, and then if that pending stake does not actually make it in, the validator would have had enough stake after all.